src/svgdom/util/finder.cpp

```diff
@@ -7,107 +7,189 @@
 using namespace svgdom;
 
 namespace{
-class CacheCreator : virtual public svgdom::const_visitor{
+class cache_creator : virtual public svgdom::const_visitor{
 public:
-	std::map<std::string, finder::element_info> cache;
+	std::unordered_map<std::string, const element*> element_by_id_cache;
+	std::unordered_map<std::string, std::list<const element*>> elements_by_class_name_cache;
```

Comment:
use `std::vector` instead of `std::list`
See https://www.youtube.com/watch?v=YQs6IC-vgmo